e2e/atlas/custom_db_roles_test.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2021 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// +build e2e atlas,generic
+
+package atlas_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+
+	"github.com/mongodb/mongocli/e2e"
+	"github.com/stretchr/testify/assert"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	listSessions   = "LIST_SESSIONS"
+	update         = "UPDATE"
+	inheritedRole  = "enableSharding@admin"
+	enableSharding = "enableSharding"
+)
+
+func TestDBRoles(t *testing.T) {
+	n, err := e2e.RandInt(1000)
+	assert.NoError(t, err)
+
+	roleName := fmt.Sprintf("role-%v", n)
+
+	cliPath, err := e2e.Bin()
+	assert.NoError(t, err)
+
+	t.Run("Create", func(t *testing.T) {
+		cmd := exec.Command(cliPath,
+			atlasEntity,
+			customDBRoleEntity,
+			"create",
+			roleName,
+			"--privilege", listSessions,
+			"--inheritedRole", inheritedRole,
+			"-o=json",
+		)
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		assert.NoError(t, err)
+
+		var role mongodbatlas.CustomDBRole
+		if err := json.Unmarshal(resp, &role); err != nil {
+			t.Fatalf("unexpected error: %v", err)
+		}
+
+		assert.Equal(t, roleName, role.RoleName)
+		assert.Len(t, role.Actions, 1)
+		assert.Equal(t, listSessions, role.Actions[0].Action)
+		assert.Len(t, role.InheritedRoles, 1)
+		assert.Equal(t, enableSharding, role.InheritedRoles[0].Role)
```

Comment:
[nit] let's do the thing where we reuse an `a` testing object

Reply:
this and the other cases have the same problem, is ok to have one `a` per test case